### PR TITLE
[@types/luxon] Lint and bring up to date.

### PR DESCRIPTION
**Pull request by mastermatt**
Even though this branch will be squatched, I've left the different changes in their own commits to aid in the review process. 

The first commit simply removes and fixes the remaining lint override rules so reviewing it [without whitespaces changes](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/72d8c50b7c0a03bbeeb824928d08dfa0d4e97e7d?w=1) is easier. 
After fixing the lint stuff, I read through the changelog and brought over the things that have been added/changed over the last couple of weeks to the main repo.
To round things out, I copied over most of the examples from the docs to the test file. The exercise did uncover a couple minor issues with local options.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - [Docs](http://moment.github.io/luxon/docs/manual/)
  - [Changelog](https://github.com/moment/luxon/blob/master/changelog.md)
- ~~[ ] Increase the version number in the header if appropriate.~~
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by mastermatt** [comment]
FYI @milesj as mentioned in #32219.

File: types/luxon/index.d.ts [hunk 1]
```diff
@@ -8,459 +8,471 @@
...
+    includeOffset?: boolean;
+    includeZone?: boolean;
+}
 
-        type ISOTimeOptions = {
-            suppressMilliseconds?: boolean;
-            suppressSeconds?: boolean;
-            includeOffset?: boolean;
-        };
+export interface ToISOTimeOptions {
```
Comment (milesj) [inline types/luxon/index.d.ts]: This rename is breaking. Maybe alias also?

File: types/luxon/index.d.ts [hunk 2]
```diff
@@ -8,459 +8,471 @@
...
-        };
+export interface ZoneOptions {
+    keepLocalTime?: boolean;
+    /**
+     * @deprecated since 0.2.12. Use keepLocalTime instead
+     */
+    keepCalendarTime?: boolean;
+}
 
-        type ToFormatOptions = DateTimeFormatOptions & {
```
Comment (milesj) [inline types/luxon/index.d.ts]: Same here about losing `ToFormatOptions`.

**Review by milesj — COMMENTED** [review]
Few Qs about breaking changes but looks good otherwise.

**Comment by mastermatt** [comment]
@milesj I've [added](f208c7b) an alias for `ISOTimeOptions`. 
I renamed it because it was being flagged by the rule `interface-name`
>  interface name must not have an "I" prefix

but apparently, it's ok for `type`s.

As for `ToFormatOptions`, that was not a simple rename. I've isolated that changeset here https://github.com/DefinitelyTyped/DefinitelyTyped/pull/32394/commits/989327292e8e2a8509a66d5625d1ee1d132da7dc which focuses on an actual change to Luxon for the v1.7.1 release. The commit message has links to the actual commits in the lib.

**Comment by milesj** [comment]
👍 LGTM

**Pull request merged**